### PR TITLE
[pt] Use new POS tagging schema

### PR DESCRIPTION
**Pull request by susanaboatto**
⚠️ This change **requires** the [portuguese-pos-dict](https://github.com/languagetool-org/portuguese-pos-dict) dependency to be `v0.13` or later!

## tl;dr

This PR adds POS tags to the `pt` tagset for enclitic pronouns.

## What changes

Let's use `diz-me` ('tell me') to illustrate the changes.

|was|will be|
|---|---|
|three tokens (`diz`, `-`, `me`)|a single token (`diz-me`)|
|each token tagged separately|tagged as `dizer[VMM02S0:PP1CSO00]`|

### Consequences
- dictionary suggestions consider whole word forms (e.g. `ama-se` <=> `amasse`);
- forms that only appear in derived environments are no longer accepted by the dictionary in isolation (e.g. `amá` and `lo` from `amá-lo` should now both be flagged as spelling errors);
- working with mesoclitics is made significantly easier (e.g. `fá-lo-á` doesn't require **five** tokens, **three** of which shouldn't exist in isolation!);
- XML rules no longer require two extra (often optional) tokens for the hyphen and pronouns.

## Also in this PR

- tokenisation logic changes from defining breaking characters to defining word characters;
- extensive changes to XML rules in all dialects to account for new tokenisation and tagging schema;
- `EnclisisFilter` and `ProclisisFilter` to make synthesising verb forms with pronouns easier.

### CODE REVIEW

**Comment by p-goulart** [comment]
@marcoagpinto I'm pinging you here so you're aware of it.

After this PR goes through, you should be able to go back to editing rules and the dictionary freely. Thank you for your patience!

**Review by p-goulart — APPROVED** [review]

**Review by jaumeortola — APPROVED** [review]

**Comment by p-goulart** [comment]
@susanaboatto I hope you've had a chance to have a look. Since it is Friday, I want to wait until Monday to merge this to avoid surprises over the weekend.
